nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ParameterContextResource.java

```diff
@@ -0,0 +1,1185 @@
+                parameterContext.authorize(authorizer, RequestAction.WRITE, user);
+
+                // Verify READ and WRITE permissions for user, for every component that is affected
+                for (final AffectedComponentEntity entity : affectedComponents) {
+                    final AffectedComponentDTO dto = entity.getComponent();
+                    if (AffectedComponentDTO.COMPONENT_TYPE_PROCESSOR.equals(dto.getReferenceType())) {
+                        final Authorizable processor = lookup.getProcessor(dto.getId()).getAuthorizable();
+                        processor.authorize(authorizer, RequestAction.READ, user);
+                        processor.authorize(authorizer, RequestAction.WRITE, user);
+                    } else if (AffectedComponentDTO.COMPONENT_TYPE_CONTROLLER_SERVICE.equals(dto.getReferenceType())) {
+                        final Authorizable service = lookup.getControllerService(dto.getId()).getAuthorizable();
+                        service.authorize(authorizer, RequestAction.READ, user);
+                        service.authorize(authorizer, RequestAction.WRITE, user);
+                    }
+                }
+            },
+            () -> {
+                // Verify Request
+                serviceFacade.verifyUpdateParameterContext(contextDto, false);
+            },
+            this::submitUpdateRequest
+        );
+    }
+
+
+    @GET
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("update-requests/{id}")
+    @ApiOperation(
+        value = "Returns the Update Request with the given ID",
+        response = ParameterContextUpdateRequestEntity.class,
+        notes = "Returns the Update Request with the given ID. Once an Update Request has been created by performing a POST to /nifi-api/parameter-contexts, "
+            + "that request can subsequently be retrieved via this endpoint, and the request that is fetched will contain the updated state, such as percent complete, the "
+            + "current state of the request, and any failures. ",
+        authorizations = {
+            @Authorization(value = "Only the user that submitted the request can get it")
+        })
+    @ApiResponses(value = {
+        @ApiResponse(code = 400, message = "NiFi was unable to complete the request because it was invalid. The request should not be retried without modification."),
+        @ApiResponse(code = 401, message = "Client could not be authenticated."),
+        @ApiResponse(code = 403, message = "Client is not authorized to make this request."),
+        @ApiResponse(code = 404, message = "The specified resource could not be found."),
+        @ApiResponse(code = 409, message = "The request was valid but NiFi was not in the appropriate state to process it. Retrying the same request later may be successful.")
+    })
+    public Response getParameterContextUpdate(@ApiParam("The ID of the Update Request") @PathParam("id") final String updateRequestId) {
```

Comment:
There is no authorization check on this endpoint. I understand that we do verify that only the user that submitted the request can retrieve it. But since the request is Parameter Context specific, we should be checking access to it. This comment applies to each async endpoint. They all appear to be missing authorization.

Reply:
Agreed.